src/pages/workspace/accounting/PolicyAccountingPage.tsx

```diff
@@ -52,9 +112,15 @@ function PolicyAccountingPage({policy, connectionSyncProgress}: PolicyAccounting
     const [threeDotsMenuPosition, setThreeDotsMenuPosition] = useState<AnchorPosition>({horizontal: 0, vertical: 0});
     const [isDisconnectModalOpen, setIsDisconnectModalOpen] = useState(false);
     const threeDotsMenuContainerRef = useRef<View>(null);
+
     const isSyncInProgress = !!connectionSyncProgress?.stageInProgress && connectionSyncProgress.stageInProgress !== CONST.POLICY.CONNECTIONS.SYNC_STAGE_NAME.JOB_DONE;
-    const policyIsConnectedToAccountingSystem = Object.keys(policy.connections ?? {}).length > 0;
-    const policyID = policy.id ?? '';
+    const connectedIntegration = accountingIntegrations.find((integration) => !!policy?.connections?.[integration]) ?? connectionSyncProgress?.connectionName;
+    const policyID = policy?.id ?? '';
+
+    const policyConnectedToXero = connectedIntegration === CONST.POLICY.CONNECTIONS.NAME.XERO;
+
+    const tenants = useMemo<Tenant[]>(() => policy?.connections?.xero?.data?.tenants ?? [], [policy?.connections?.xero.data.tenants]);
```

Comment:
Another crash on disconnecting account - 
1. Connect Xero
2. From the three dot menu at top right, click disconnect.
3. App crash

<img width="1512" alt="image" src="https://github.com/Expensify/App/assets/29673073/9b59be2a-7804-4c8f-9566-1831fd6decfb">
<img width="719" alt="image" src="https://github.com/Expensify/App/assets/29673073/cce0ae46-4595-4655-b6a3-b6f698ed9cb4">